src/azure/azure-mesh-adapter.ts

```diff
@@ -110,16 +114,89 @@ export class AzureMeshAdapter implements MeshAdapter {
           platform: MeshPlatform.Azure,
           nativeObj: sub,
           costs: [],
+          roleAssignments: [],
         };
       }),
     );
   }
 
+  async loadTenantRoleAssignments(tenants: MeshTenant[]): Promise<void> {
+    // Only work on Azure tenants
+    const azureTenants = tenants.filter((t) => isSubscription(t.nativeObj));
+
+    for (const t of azureTenants) {
+      const roleAssignments = await this.loadRoleAssignmentsForTenant(t);
+      t.roleAssignments.push(...roleAssignments);
+    }
+  }
+
+  private async loadRoleAssignmentsForTenant(
+    tenant: MeshTenant,
+  ): Promise<MeshTenantRoleAssignment[]> {
+    if (!isSubscription(tenant.nativeObj)) {
```

Comment:
d: in theory not required as we already filtered it before. But is handy because then the nativeObj is considered a Subscription and some extra safety is also never wrong... but not sure an `as Subscription` would also do the trick. :D 
t: also a MeshAzureError is probably better suited here.

Reply:
As unnecessary as this is, I'd always prefer not having to cast (`as ...`).